src/main/scala/junctions/asyncqueue.scala

```diff
@@ -66,9 +75,13 @@ class AsyncQueueSink[T <: Data](gen: T, depth: Int, sync: Int, clockIn: Clock, r
   // On an FPGA, only one input changes at a time => mem updates don't cause glitches
   // The register only latches when the selected valued is not being written
   val index = if (depth == 1) UInt(0) else ridx(bits-1, 0) ^ (ridx(bits, bits) << (bits-1))
-  io.deq.bits  := RegEnable(io.mem(index), valid && !reset)
-  io.deq.valid := RegNext(valid, Bool(false))
-  io.ridx := RegNext(ridx, UInt(0))
+  io.deq.bits  := RegEnable(io.mem(index), valid && !reset) // This does NOT need to be reset.
```

Comment:
@terpstra I removed this reset. Let me know if there was something I was missing there.